docs/src/content/en/reference/auth/auth0.mdx

```diff
@@ -0,0 +1,131 @@
+---
+title: "MastraAuthAuth0 Class"
+description: "API reference for the MastraAuthAuth0 class, which authenticates Mastra applications using Auth0 authentication."
+---
+
+# MastraAuthAuth0 Class
+
+The `MastraAuthAuth0` class provides authentication for Mastra using Auth0. It verifies incoming requests using Auth0-issued JWT tokens and integrates with the Mastra server using the `experimental_auth` option.
+
+## Usage examples
+
+### Basic usage with environment variables
+
+```typescript filename="src/mastra/index.ts" showLineNumbers copy
+import { Mastra } from "@mastra/core/mastra";
+import { MastraAuthAuth0 } from '@mastra/auth-auth0';
+
+// Automatically uses AUTH0_DOMAIN and AUTH0_AUDIENCE env vars
```

Comment:
I'd pull this explanation out of the code snippet and add it under `## Usage examples`